internal/mode/static/state/resolver/resolver.go

```diff
@@ -46,6 +54,7 @@ func (e *ServiceResolverImpl) Resolve(
 	ctx context.Context,
 	svcNsName types.NamespacedName,
 	svcPort v1.ServicePort,
+	ipv4Enabled, ipv6Enabled bool,
```

Comment:
typically, using bools as an argument is an anti-pattern

here, we use two `ipv4Enabled, ipv6Enabled bool` to model supported addresses with 3 combinations allowed: TF. FT. TT

instead of such modeling, I suggest passing `to` Resolve and `resolveEndpoints` a list of  supported address types:

```
addressTypes []discoveryV1.AddressType
```

for example, in case of dual stack, we pass `[]discoveryV1.AddressType{ discoveryV1.AddressTypeIPv4,  discoveryV1.AddressTypeIPv6}`

as the benefit, filtering logic in `resolveEndpoints` will also get simplified, as you will just compare the `eps.AddressType` with elements of `addressTypes`.